src/Components/Shared/src/WebEventData/WebEventData.cs

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using Microsoft.AspNetCore.Components.RenderTree;
+
+namespace Microsoft.AspNetCore.Components.Web
+{
+    internal class WebEventData
+    {
+        // This class represents the second half of parsing incoming event data,
+        // once the event ID (and possibly the type of the eventArgs) becomes known.
+        public static WebEventData Parse(
+            Renderer renderer,
+            JsonSerializerOptions jsonSerializerOptions,
+            JsonElement eventDescriptorJson,
```

Comment:
Silly question - but why not base these on JsonReader instead of JsonElement?

Reply:
There was one code path (among 3 that use this) that already constructed a `JsonElement`, so it was easier to use this than to do a bigger refactoring. there.

Reply:
Got it. Thanks for the answer.

If we were really looking to squeeze out the perf, I wonder how much more could be gained to refactor it to straight `Utf8JsonReader`. But if we already have acceptable perf, it is probably not worth it.